crates/polars-ops/src/series/ops/is_in.rs

```diff
@@ -556,6 +560,16 @@ fn is_in_struct(ca_in: &StructChunked, other: &Series) -> PolarsResult<BooleanCh
     }
 }
 
+#[cfg(feature = "dtype-categorical")]
+fn is_in_string_categorical(
+    ca_in: &StringChunked,
+    other: &CategoricalChunked,
+) -> PolarsResult<BooleanChunked> {
+    // We need only check against the unique categories used by this array.
+    let other: StringChunked = other.unique().unwrap().iter_str().collect_trusted();
```

Comment:
I am not happy with a double collection here. If the Categorical supports `FAST_UNIQUE` we can directly get the categories out. And otherwise we can create a `HashSet` and probably factor out some of the binary code.

Reply:
@ritchie46 it looks like `can_fast_unique` isn't exposed outside of polars-core, so I don't think we can distinguish the fast unique path without pushing some of this logic to polars-core. Is that what you would recommend here, or is there a better path that I'm not seeing?

Alternatively we could just expose `can_fast_unique` which, at least in this case, shows that it may be useful outside of core.

Reply:
It can be exposed via private `_undersored` methods if really needed.